run-perf.pl

```diff
@@ -0,0 +1,442 @@
+	my ($test_name, $version, $thread_count, $cert_dir) = @_;
+	my $test_cmd = get_tool($test_name, $version);
+
+	if (! $test_cmd) {
+		#
+		# tools was not for desired test,
+		# are we dealing with for example rwlocks_wlock?
+		#
+		my $tool_name = $test_name;
+
+		#
+		# we use underscore ('_') in test names as a delimiter
+		# between test program tool and eventual parametrs
+		# to test (for example rwlocks_rlock).
+		#
+		# evp_setpeer test tool clashes with our convention,
+		# so we are dealing with special case
+		#
+		if ($tool_name =~ /evp_setpeer/) {
+			$tool_name = "evp_setpeer";
+			$test_cmd = get_tool($tool_name, $version);
+		} else {
+			$tool_name = ( split(/_/, $test_name) )[0];
+		}
+
+		$test_cmd = get_tool($tool_name, $version);
+		if (! $test_cmd) {
+			print "No tool for $tool_name ($version)\n";
+			return undef;
+		}
+
+		if ("$tool_name" eq "pkeyread") {
+			my $cipher;
+			my $format;
+
+			($tool_name, $cipher, $format) = split(/_/, $test_name);
+			$test_cmd = get_tool($tool_name, $version);
+			$test_cmd = "$test_cmd -k $cipher -f $format ";
+		} elsif ("$tool_name" eq "evp_setpeer") {
+			my $cipher;
+
+			$cipher = ( split(/_/, $test_name) )[2];
+			$test_cmd = get_tool($tool_name, $version);
+			$test_cmd = "$test_cmd -k $cipher";
+		}
+	}
+
+	#
+	# add common options/arguments
+	#	-t, terse output
+	#	number of threads to use
+	#	directory to certificates (when needed)
+	#
+	if ($test_name == "handshake" || $test_name == "x509storeissuer") {
+		$test_cmd = "$test_cmd -t $cert_dir $thread_count";
+	} else {
+		$test_cmd = "$test_cmd -t $thread_count";
+	}
+
+	return $test_cmd
+}
+
+my sub do_test {
+	my ($test_name, $version, $thread_count, $cert_dir) = @_;
+	my $test_cmd;
+	my $result;
+
+	$test_cmd = get_test_cmd($test_name, $version, $thread_count,
+	    $cert_dir);
+	if (! $test_cmd) {
+		return undef;
+	}
+
+	$ENV{EVP_FETCH_TYPE} = "MD:MD5";
+	if ("$^O" eq "MSWin32") {
+		#
+		# There is no LD_LIBRARY_PATH, nor RTPATH on windows.
+		# We have two options here:
+		# (https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order)
+		#	1) copy .dll files to the directory where tools
+		#	reside
+		#
+		#	2) add path .dll to PATH env. variable
+		# We opt for 2).
+		#
+		my $path_to_dll = File::Spec->catdir($OPENSSL_BINARIES,
+		    join('-', 'openssl', $version), 'bin');
+		$ENV{PATH} = "%PATH;$path_to_dll";
+	}
+	open(PIPE, "$test_cmd |") || return undef;
+	while (<PIPE>) {
+		chomp;
+		$result = $_;
+	}
+	close PIPE;
+
+	#
+	# need to port-process output for those tests
+	#
+	if ("$test_name" eq "rwlocks_wlock") {
+		#
+		# rwlocks test returns two numbers,
+		# the first is for write-lock
+		#
+		$result = ( split(/ /, $result) )[0];
+	} elsif ("$test_name" eq "rwlocks_rlock") {
+		#
+		# the second number is for read lock
+		#
+		$result = ( split(/ /, $result) )[1];
+	}
+
+	if ($result =~ "nan") {
+		#
+		# running 'rwlocks -t 1' gives output similar to this:
```

Comment:
hmm, that seems like a bug in the locks test, should probably open a separate issue for this to require t be at least 2

Reply:
I need to talk about it with Dima. His automation expects single number as output in terse mode. It makes
his life simple. I'm inclining towards introducing option '-w' so tool will report stats for write-lock. if '-w' will not be present the tool will report stats for readlocks. I think this should be fine for him.

I agree this can be done as a follow-up PR.